app/src/main/java/com/example/lunark/adapters/ReservationsCancelListAdapter.java

```diff
@@ -0,0 +1,203 @@
+        return new ReservationsCancelListAdapter.ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ReservationsCancelListAdapter.ViewHolder holder, int position) {
+        Reservation reservation = reservations.get(position);
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd MMM yyyy", Locale.getDefault());
+        Long userId = reservation.getGuestId();
+        Long reservationId = reservation.getId();
+        Long propertyId = reservation.getPropertyId();
+
+        holder.dateRange.setText(dateFormat.format(reservation.getStartDate()) + " - " + dateFormat.format(reservation.getEndDate()));
+        holder.price.setText("Price: $" + String.valueOf(reservation.getPrice()));
+
+        fetchPropertyData(propertyId, new ReservationsCancelListAdapter.PropertyDataCallback() {
+            @Override
+            public void onPropertyDataFetched(Property property) {
+                if (property.getImages().size() > 0) {
+                    PropertyImage image = property.getImages().get(0);
+                    Glide.with(fragment)
+                            .load(ClientUtils.SERVICE_API_PATH + "properties/" + property.getId() + "/images/" + image.getId())
+                            .into(holder.getPropertyImage());
+                }
+                holder.getGuestName().setText(reservation.getPropertyName());
+                holder.getPropertyName().setText(property.getAddress().getCity().toString() + ", " + property.getAddress().getCity().toString());
+            }
+
+            @Override
+            public void onPropertyDataFetchFailed() {
+            }
+        });
+
+        holder.getBtnCancel().setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                cancelReservation(reservationId);
+                Toast.makeText(fragment.getContext(), new String("Reservation canceled!"), Toast.LENGTH_SHORT).show();
+                reservations.remove(holder.getAdapterPosition());
+                notifyItemRemoved(holder.getAdapterPosition());
+            }
+        });
+
+
+    }
+    private void fetchUserData(Long userId, final ReservationsCancelListAdapter.UserDataCallback callback) {
+        Call<AccountDto> call = ClientUtils.accountService.getAccount(userId);
+        call.enqueue(new Callback<AccountDto>() {
+            @Override
+            public void onResponse(Call<AccountDto> call, Response<AccountDto> response) {
+                if (response.isSuccessful() && response.body() != null) {
+                    AccountDto accountDto = response.body();
+                    callback.onUserDataFetched(accountDto);
+                } else {
+                    callback.onUserDataFetchFailed();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<AccountDto> call, Throwable t) {
+                callback.onUserDataFetchFailed();
+            }
+        });
+    }
```

Comment:
This method appears to be unused.
```suggestion
    private void fetchUserData(Long userId, final ReservationsCancelListAdapter.UserDataCallback callback) {
        Call<AccountDto> call = ClientUtils.accountService.getAccount(userId);
        call.enqueue(new Callback<AccountDto>() {
            @Override
            public void onResponse(Call<AccountDto> call, Response<AccountDto> response) {
                if (response.isSuccessful() && response.body() != null) {
                    AccountDto accountDto = response.body();
                    callback.onUserDataFetched(accountDto);
                } else {
                    callback.onUserDataFetchFailed();
                }
            }

            @Override
            public void onFailure(Call<AccountDto> call, Throwable t) {
                callback.onUserDataFetchFailed();
            }
        });
    }
```